java/ql/src/Security/CWE/CWE-502/UnsafeDeserialization.ql

```diff
@@ -54,6 +66,33 @@ class UnsafeDeserializationConfig extends TaintTracking::Configuration {
       ma.getArgument(0) = node.asExpr() and
       exists(SafeJsonIoConfig sji | sji.hasFlowToExpr(ma.getArgument(1)))
     )
+    or
+    exists(MethodAccess ma |
+      (
+        ma.getMethod() instanceof FlexjsonDeserializeMethod or
+        ma.getMethod() instanceof JoddJsonParseMethod
```

Comment:
I note there are various JoddJson parse methods that take `Class` arguments other than arg1, e.g. parseAsMap. Should we just check for any argument whose type is `java.lang.Class`?

Reply:
Changed

Reply:
This does not seem to have changed: it still only checks argument 1.

Also my question is still outstanding: why do you care about `isFinal()`? When could a user call `parse(char[] input, java.lang.Class<T> targetType)` and introduce an unsafe deserialization vulnerability?